.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow will generate the static page under `pr` subdirectory inside the `gh-pages` branch
+
+# This workflow will run every time there's a PR opened, reopened, synchronize, or closed
+
+name: Deploy PR previews
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - closed
+
+concurrency: preview-${{ github.ref }}
+
+jobs:
+  deploy-preview:
+    runs-on: ubuntu-20.04
```

Comment:
I notice that we're specifying `ubuntu-20.04` here and `ubuntu-latest` above. would it make sense to have these be consistent?